corehq/apps/data_interfaces/templates/data_interfaces/partials/case_rule_criteria.html

```diff
@@ -21,35 +21,34 @@
     {% crispy criteria_form criteria_form.case_type_helper %}
     <div data-bind="template: {name: get_ko_template_id, foreach: criteria, afterRender:disableCriteriaField}"></div>
     <div class="form-group" data-bind="template: {afterRender:disableCriteriaField}">
-        <div class="alert alert-block alert-danger" data-bind="visible: show_add_filter_warning() && selected_case_filter_id() !== 'select-one'">
-            <span class="help-block"><strong>{% trans "Please click Add filter or clear the dropdown" %}</strong></span>
-        </div>
-        <div class="col-xs-3 col-xs-offset-3">
-            <select class="select form-control" data-bind="value: selected_case_filter_id">
-                <option value="select-one">{% trans "(Select one and click Add filter)" %}</option>
-                {% if form.allow_case_modified_filter %}
-                <option value="case-modified-filter">{% trans "Case not modified since" %}</option>
-                {% endif %}
-                {% if form.allow_case_property_filter %}
-                <option value="case-property-filter">{% trans "Case property" %}</option>
-                {% endif %}
-                {% if form.allow_date_case_property_filter %}
-                <option value="date-case-property-filter">{% trans "Date case property" %}</option>
-                <option value="advanced-date-case-property-filter">{% trans "Date case property (advanced)" %}</option>
-                {% endif %}
-                {% if form.allow_parent_case_references %}
-                <option value="parent-closed-filter">{% trans "Parent case is closed" %}</option>
-                {% endif %}
-                {% if form.is_system_admin %}
-                <option value="custom-filter">{% trans "Custom filter" %}</option>
-                {% endif %}
-            </select>
-        </div>
-        <div class="col-xs-2">
-            <button class="btn btn-success" data-bind="click: add_filter">
-                <i class="fa fa-plus"></i>
-                {% trans "Add filter" %}
-            </button>
+        <div class="col-xs-5 col-xs-offset-3">
+            <div class="btn-group">
+                <button class="btn btn-default" type="button" data-toggle="dropdown">
+                    <i class="fa fa-plus"></i>
+                    {% trans "Select a filter" %}
+                </button>
+                <button class="btn btn-default dropdown-toggle" data-toggle="dropdown">
+                    <span class="caret"></span>
+                </button>
+                <ul class="dropdown-menu">
+                    {% if form.allow_case_modified_filter %}
+                    <li data-bind="click: function() { add_filter('case-modified-filter'); }"><a>{% trans "Case not modified since" %}</a></li>
```

Comment:
adding event using js/jquery to all `<li>s` with class and data with data-x attribute would be much cleaner. 
I wonder doing that would require render after or not!. If so, then would require it to add in at line 23.

Reply:
I don't agree that would be cleaner - mixing DOM element selection & event attaching with knockout bindings is something to avoid. When you're working in knockout, the js should be concerned just with the data model, and it's the HTML where you control which elements are bound to which parts of that data model.